packages/core/src/telemetry/startupProfiler.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { performance } from 'node:perf_hooks';
```

Comment:
I think this API is very rich and can help make your implementation a bit more robust. See mark() and measure().